Helper/Order.php

```diff
@@ -511,14 +513,21 @@ protected function setAddress($quoteAddress, $address)
      */
     protected function setShippingAddress($quote, $transaction)
     {
-        $address = $transaction->order->cart->shipments[0]->shipping_address ?? null;
-        $referenceShipmentMethod = $transaction->order->cart->shipments[0]->reference ?? null;
-
-        if ($address) {
-            $this->setAddress($quote->getShippingAddress(), $address);
-            if (isset($referenceShipmentMethod) && $this->configHelper->isPickupInStoreShippingMethodCode($referenceShipmentMethod)) {
-                $addressData = $this->configHelper->getPickupAddressData();
-                $quote->getShippingAddress()->addData($addressData);
+        if (isset($transaction->order->cart->in_store_shipments)) {
+            $address = $transaction->order->cart->in_store_shipments[0]->shipment->shipping_address ?? null;
```

Comment:
can we add check if `$transaction->order->cart->in_store_shipments[0]->shipment->shipping_address` set?

Reply:
Done